src/js/webSerial.js

```diff
@@ -30,12 +30,32 @@ class WebSerial extends EventTarget {
 
         this.logHead = "SERIAL: ";
 
+        this.port_counter = 0;
+        this.ports = null;
         this.port = null;
         this.reader = null;
         this.writer = null;
         this.reading = false;
 
         this.connect = this.connect.bind(this);
+
+        navigator.serial.addEventListener("connect", e => this.handleNewDevice(e.target).bind(this));
+        navigator.serial.addEventListener("disconnect", e => this.handleRemovedDevice(e.target).bind(this));
+    }
+
+    handleNewDevice(device) {
+        const added = this.createPort(device);
+        if (this.ports === null) {
+            this.ports = [];
+        }
```

Comment:
small nitpick, but we could just initialise it to empty array in the first place. (added suggestion above)

Reply:
Yes, the reason was explained in the earlier comment, but I will change it.

Reply:
Fixed.